test/integration/api/offline-entities.js

```diff
@@ -609,4 +609,135 @@ describe('Offline Entities', () => {
         });
     }));
   });
+
+  describe('reprocessing submissions when toggling approvalRequired dataset flag', () => {
+    it('should not over-process a submission that is being held because it is later in a run', testOfflineEntities(async (service, container) => {
+      const asAlice = await service.login('alice');
+
+      // Configure the entity list to create entities on submission approval
+      await asAlice.patch('/v1/projects/1/datasets/people')
+        .send({ approvalRequired: true })
+        .expect(200);
+
+      // Create a submission that would create an entity on approval. Don't approve it yet.
+      // trunk version is 1, but base version is higher than trunk version indicating it is later in the run
+      await asAlice.post('/v1/projects/1/forms/offlineEntity/submissions')
+        .send(testData.instances.offlineEntity.one
+          .replace('branchId=""', `branchId="${uuid()}"`)
+          .replace('baseVersion="1"', 'baseVersion="2"')
+        )
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+
+      await exhaust(container);
+
+      // Observe that the submission was initially processed without error
+      await asAlice.get('/v1/projects/1/forms/offlineEntity/submissions/one/audits')
+        .expect(200)
+        .then(({ body }) => {
+          should.not.exist(body[0].details.problem);
+        });
+
+      // Observe there is no second version yet this update was not applied
+      await asAlice.get('/v1/projects/1/datasets/people/entities/12345678-1234-4123-8234-123456789abc/versions')
+        .then(({ body: versions }) => {
+          versions.length.should.equal(1);
+        });
+
+      // Trigger the submission being reprocessed by updating the entity list
+      await asAlice.patch('/v1/projects/1/datasets/people?convert=true')
+        .send({ approvalRequired: false })
+        .expect(200);
+
+      await exhaust(container);
+
+      await asAlice.get('/v1/projects/1/datasets/people/entities/12345678-1234-4123-8234-123456789abc/versions')
+        .then(({ body: versions }) => {
+          versions.length.should.equal(1);
+        });
+
+      await asAlice.get('/v1/projects/1/forms/offlineEntity/submissions/one/audits')
+        .expect(200)
+        .then(({ body }) => {
+          body.length.should.equal(1);
+          should.not.exist(body[0].details.problem);
+        });
```

Comment:
Check in DB that number of submissions in backlog = 1

Send in missing submission and see that backlog goes to 0